MonstieWash/Assets/Scripts/Decoration/DecorationManager.cs

```diff
@@ -0,0 +1,523 @@
+        {            
+            m_currentlyHeldDecoration.placeInScene();
+            RefreshBarUI();
+
+            managerStatus = ManagerStatus.EmptyHand;
+            m_currentlyHeldDecoration = null;
+        }
+    }
+
+    /// <summary>
+    /// Starts coroutine that handles monster mood management.
+    /// </summary>
+    private void LockMonsterMood()
+    {
+        StartCoroutine(HandleMonsterMoods());
+    }
+
+    /// <summary>
+    /// Locks monster to a single chosen mood for the duration of decorating.
+    /// </summary>
+    /// <returns></returns>
+    private IEnumerator HandleMonsterMoods()
+    {
+        MonsterBrain mb = FindFirstObjectByType<MonsterBrain>();
+
+        mb.UpdateMood(99999, decoMood);
+        foreach (MoodType mt in mb.Moods)
+        {
+            if (mt != decoMood)
+            {
+                mb.UpdateMood(-99999, mt);
+            }
+
+        }
+        
+
+        //Wait for frame to finish so animation updates can take place based on new moods.
+        yield return new WaitForEndOfFrame();
+
+        mb.PauseBrain(true);
+    }
+
+    /// <summary>
+    /// Calls screenshot coroutine.
+    /// </summary>
+    public void TakePolaroid()
+    {
+        StartCoroutine(PolaroidSnap());
+    }
+
+    /// <summary>
+    /// Takes a screenshot in an area and saves it to file. Default path will be User/AppData/LocalLow/MagicalCleaningAgency/MonstieWash/variable name chosen.
+    /// </summary>
+    /// <returns></returns>
+    private IEnumerator PolaroidSnap()
+    {       
+        //Screenshot area definition
+        Vector3[] corners = new Vector3[4];
+        screenshotArea.GetWorldCorners(corners);
+        int width = ((int)corners[3].x - (int)corners[0].x) - 100;
+        int height = (int)corners[1].y - (int)corners[0].y;
+        var startX = corners[0].x;
+        var startY = corners[0].y;
+
+        //Turn off bar and button for photo.
+        this.transform.GetChild(0).gameObject.SetActive(false);
+        FindFirstObjectByType<FinishLevelButton>().gameObject.SetActive(false);
+        m_hand.gameObject.SetActive(false);
+
+        //Wait for seconds and camera flash + sound here.
+        camFlash.FadeIn();
+        yield return new WaitForSeconds(1f);
+        
+
+        //Wait for frame to fully render and update before taking screenshot.
+        yield return new WaitForEndOfFrame();
+
+        //Apply screen view to a texture (Take screenshot)
+        Texture2D tempText = new Texture2D(width, height, TextureFormat.RGB24, false);
+        tempText.ReadPixels(new Rect(startX, startY, width, height), 0, 0);
+        tempText.Apply();
+
+        //Apply a modifier to end of file based on count of objects stored.
+        var fileCount = Directory.GetFiles(Application.persistentDataPath).Length;
+        //Save to file.
+        byte[] byteArray = tempText.EncodeToPNG();
+        string saveLocation = Application.persistentDataPath + savePath + fileCount + ".Png";
```

Comment:
### 🗒️ **Convention** | _This is for maintaining our coding conventions._

Can make all of these into `var`'s.

```suggestion
        var corners = new Vector3[4];
        screenshotArea.GetWorldCorners(corners);
        var width = ((int)corners[3].x - (int)corners[0].x) - 100;
        var height = (int)corners[1].y - (int)corners[0].y;
        var startX = corners[0].x;
        var startY = corners[0].y;

        //Turn off bar and button for photo.
        this.transform.GetChild(0).gameObject.SetActive(false);
        FindFirstObjectByType<FinishLevelButton>().gameObject.SetActive(false);
        m_hand.gameObject.SetActive(false);

        //Wait for seconds and camera flash + sound here.
        camFlash.FadeIn();
        yield return new WaitForSeconds(1f);
        

        //Wait for frame to fully render and update before taking screenshot.
        yield return new WaitForEndOfFrame();

        //Apply screen view to a texture (Take screenshot)
        var tempText = new Texture2D(width, height, TextureFormat.RGB24, false);
        tempText.ReadPixels(new Rect(startX, startY, width, height), 0, 0);
        tempText.Apply();

        //Apply a modifier to end of file based on count of objects stored.
        var fileCount = Directory.GetFiles(Application.persistentDataPath).Length;
        //Save to file.
        var byteArray = tempText.EncodeToPNG();
        var saveLocation = Application.persistentDataPath + savePath + fileCount + ".Png";
```